tofu/geom/_core_optics.py

```diff
@@ -2090,9 +2123,19 @@ def plot_focal_error_summed(
                         xi=xi, xj=xj,
                         det=det,
                         err=err,
+                        lambda_interval_min=lambda_interval_min,
+                        lambda_interval_max=lambda_interval_max,
                         plot=False,
                     )[0],
                 )
+                test_lamb_interv[jj, ii] = self.calc_johannerror(
+                    xi=xi, xj=xj,
+                    det=det,
+                    err=err,
+                    lambda_interval_min=lambda_interval_min,
+                    lambda_interval_max=lambda_interval_max,
+                    plot=False,
+                )[4]
```

Comment:
Good but:

* here you call the method twice (first to compute the mean erro, then to get the interval bool).
* Executing the method is the longest task => call it once, store the result to extract the 2 quantities at once

```
error_lambda_temp, test_lamb_interv[jj, ii] = 
                    self.calc_johannerror(
                        xi=xi, xj=xj,
                        det=det,
                        err=err,
                        lambda_interval_min=lambda_interval_min,
                        lambda_interval_max=lambda_interval_max,
                        plot=False,
                    )[::4]
     error_lambda[ii, jj] = np.nanmean(error_lambda_temp)
```


This syntax (`[::5]`) means 'take one every 5 indices":
So if you want to take only the first and last output, you can use `[::n]`, where `n = len(list) - 1`

```
In [5]: ll = ['a', 'b', 'c', 'd', 'e']                                                                                                               

In [6]: ll[::2]                                                                                                                                      
Out[6]: ['a', 'c', 'e']

In [7]: ll[::3]                                                                                                                                      
Out[7]: ['a', 'd']

In [8]: ll[::4]                                                                                                                                      
Out[8]: ['a', 'e']

```